File: benchmarks/python/08_collatz.py
def collatz_len(n):
    count = 0
    x = n
    while x != 1:
        if x % 2 == 0:
            x = x // 2
        else:
            x = 3 * x + 1
        count += 1
    return count

def longest_collatz(limit):
    best = 0
    for n in range(2, limit):
        l = collatz_len(n)
        if l > best:
            best = l
    return best
print(longest_collatz(1000000))

File: benchmarks/node/08_collatz.mjs
function collatz_len(n) {
    let count = 0, x = n;
    while (x !== 1) {
        x = x % 2 === 0 ? x / 2 : 3 * x + 1;
        count++;
    }
    return count;
}
function longest_collatz(limit) {
    let best = 0;
    for (let n = 2; n < limit; n++) {
        const l = collatz_len(n);
        if (l > best) best = l;
    }
    return best;
}
console.log(longest_collatz(1000000));
